README.md
# Open Cancer TherApeutic Discovery (OCTAD) database package

### Package overview
As the field of precision medicine progresses, we start to tailor treatments for cancer patients classified not only by their clinical, but also by their molecular features. The emerging cancer subtypes defined by these features require dedicated resources to assist the discovery of drug candidates for preclinical evaluation. Voluminous cancer patient gene expression profiles have been accumulated in public databases, enabling the creation of a cancer-specific expression signature. Meanwhile, large-scale gene expression profiles of chemical compounds became available in recent years. By matching the cancer-specific expression signature to compound-induced gene expression profiles from large drug libraries, researchers can prioritize small molecules that present high potency to reverse expression of signature genes for further experimental testing of their efficacy. This approach has proven to be an efficient and cost-effective way to identify efficacious drug candidates. However, the success of this approach requires multiscale procedures, imposing significant challenges to many labs. Therefore, we present OCTAD (http://octad.org): an open workspace for virtually screening compounds targeting precise cancer patient groups using gene expression features. We have included 19,127 patient tissue samples covering 50 cancer types, and expression profiles for 12,442 distinct compounds.  We will continue to include more tissue samples. We streamline all the procedures including deep-learning based reference tissue selection, disease gene expression signature creation, drug reversal potency scoring, and in silico validation. We release OCTAD as a web portal and a standalone R package to allow experimental and computational scientists to easily navigate the tool. The code and data can also be employed to answer various biological questions.

### How to install
# Bioconductor installation:
```{r eval=FALSE} 
if (!requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager")
BiocManager::install("octad.db")
```
The code can be viewed at the GitHub repository, which also lists the contributor code of conduct:
\url{https://github.com/Bin-Chen-Lab/OCTAD}

# Please note, this is a support package for the main package ```octad``` which can be obtained [there](https://github.com/Bin-Chen-Lab/octad)

###  Contacts and citation
  If you use our work, please cite the [OCTAD paper](https://www.nature.com/articles/s41596-020-00430-z). Both OCTAD package and website was developed by [Bin Chen laboratory](http://binchenlab.org/).
Examples and questions can be addressed to Evgenii Chekalin, PhD, chekali1@msu.edu or Bin Chen, PhD, PI, bin.chen@hc.msu.edu


NEWS.md
# octad.db 0.99.22

* Added a `NEWS.md` file to track changes to the package.

# octad.db 0.99.23

* Submitted to Bioconductor


R/zzz.R
#' @export
#' @importFrom ExperimentHub ExperimentHub

#.my_internal_global_variables <- new.env(parent=emptyenv())

#.get_eh <- function() get("eh", envir=.my_internal_global_variables)
#.set_eh <- function(value) assign("eh", value,
#                                  envir=.my_internal_global_variables)

#toto <- function()
#{
#  eh <- try(.get_eh(), silent=TRUE)
#  if (inherits(eh, "try-error")) {
#    eh <- ExperimentHub()
#    .set_eh(eh)
#  }
#  eh
#}
.eh <- NULL

.onLoad <- function(libname, pkgname)
{
  .eh <<- ExperimentHub::ExperimentHub()
}

.onAttach <- function(libname, pkgname) {
  packageStartupMessage("Welcome to octad.db package. This is a database package, pipeline available via the octad package. If you want to run the pipeline on the webserver, please, refer to octad.org")
}


R/octad.db.R
#' Open Cancer TherApeutic Discovery (OCTAD) database package
#'
#' This is a support package for the main package ```octad``` which can be obtained [here](https://bioconductor.org/packages/octad)
#' 
#'Package includes all required data for drug repurposing OCTAD pipeline. 
#'Initialization of the pipeline starts with listing all available samples:
#'
#'```phenoDF=.eh[["EH7274"]]``` #load data.frame with samples included in the OCTAD database. 
#'```head(phenoDF)``` #list all data included within the package
#'```
#'Besides, the package includes examples of the output from diffExp and runsRGES functions from the octad package along with desctiption of the data:
#'```res=octad.db::res_example``` #load example res from octad.db
#'```?res_example```
#'```sRGES=octad.db::sRGES_example``` #load example sRGES from octad.db
#'```?sRGES_example```
#'
#'
#'
#'
#' The code can be viewed at the GitHub repository,
#' which also lists the contributor code of conduct:
#'
#' \url{https://bioconductor.org/packages/octad} or \url{https://github.com/Bin-Chen-Lab/OCTAD} for the pipeline package
#' 
#' 
#' @references
#' Zeng, B., Glicksberg, B.S., Newbury, P., Chekalin, E., Xing, J., Liu, K., Wen, A., Chow, C. and Chen, B., 2021. OCTAD: an open workspace for virtually screening therapeutics targeting precise cancer patient groups using gene expression features. Nature protocols, 16(2), pp.728-753.
#' [https://www.nature.com/articles/s41596-020-00430-z](https://www.nature.com/articles/s41596-020-00430-z)
#' @docType package
#' @name octad.db
NULL

inst/scripts/make-metadata.R
meta <- data.frame(Title=c('CCLE.log2.read.count.matrix',
                           'CCLE.overlaps',
                           'CCLE.sample.meta',
                           'CTRPv2.sensprof',
                           'EncoderDF',
                           'cmpd_sets_ChemCluster',
                           'cmpd_sets_chembl_targets',
                           'cmpd_sets_mesh',
                           'fda_drugs',
                           'lincs_sig_info',
                           'lincs_signatures',
                           'merged_gene_info',
                           'octad.LINCS.counts',
                           'phenoDF',
                           'random_gsea_score',
                           'tsne',
                           'octad.counts.and.tpm.h5',
                           'res_example.rda',
                           'sRGES_example.rda'
),
Description=c(
'log2-transformed matrix containing expression of 56,318 genes in 1,019 cell lines provided in CCLE database',
'log2-transformed matrix containing expression of 56,318 genes in 51 cell lines present in both LINCS and CCLE databases',
'data.frame containing cell lines name and tissue of origin from CCLE database',
'Cancer Therapeutics Response Portal (CTRP) data.frame containing AUC and IC50 for every drug-cell line pair',
'a data.frame (64 features by 19127 samples) computed from autoEncoder for every sample in the OCTAD database',
'list of compounds and their clusters based on chemical structures',
'list of compounds and their targets presented in ChEMBL database',
'list of compounds and their associations with 662 pharmacological MeSH Terms provided in PubChem',
'data.frame of FDA approved drugs containing name, target and clinical phase',
'data.frame for LINCS signatures including experiment id, cell line used in the experiment, purturbagen name, purturbagen type, dose and time',
'Differential expression of 978 genes in 416,560 experiments',
'data.frame with the annotation of 75,078 genes which could be used for identifier mapping',
'expression matrix of 965 genes and 19127 samples in OCTAD database (965 genes were profiled in LINCS)',
'data.frame with samples from the OCTAD database including tissue of origin, sample type (tumor-derived or healthy-derived), cancer, orginal source (e.g. TCGA), mutation and stage data if available',
'precomputed permutated gsea scores for chembl, mesh and ChemCluster databases that are used for drug enrichment analysis',
'2 dimension tSNE for every sample in the OCTAD database along with sample.id, cancer type and source of the initial data',
'expression (TPM and raw counts) matrix of 60,498 genes and 19,127 samples',
'example of differential expression analysis of HCC vs adjacent liver for OCTAD performed on genes contained in LINCs database',
'example of RGES computed from DE analysis of HCC vs adjacent liver for OCTAD performed on genes contained in LINCs database'
),
BiocVersion='3.15',
Genome='hg19',
SourceType='RDA',
SourceUrl=c(
  'https://chenlab-data-public.s3.amazonaws.com/octad/octad.db/CCLE.log2.read.count.matrix.rda',
  'https://chenlab-data-public.s3.amazonaws.com/octad/octad.db/CCLE.overlaps.rda',
  'https://chenlab-data-public.s3.amazonaws.com/octad/octad.db/CCLE.sample.meta.rda',
  'https://chenlab-data-public.s3.amazonaws.com/octad/octad.db/CTRPv2.sensprof.rda',
  'https://chenlab-data-public.s3.amazonaws.com/octad/octad.db/EncoderDF.rda',
  'https://chenlab-data-public.s3.amazonaws.com/octad/octad.db/cmpd_sets_ChemCluster.rda',
  'https://chenlab-data-public.s3.amazonaws.com/octad/octad.db/cmpd_sets_chembl_targets.rda',
  'https://chenlab-data-public.s3.amazonaws.com/octad/octad.db/cmpd_sets_mesh.rda',
  'https://chenlab-data-public.s3.amazonaws.com/octad/octad.db/fda_drugs.rda',
  'https://chenlab-data-public.s3.amazonaws.com/octad/octad.db/lincs_sig_info.rda',
  'https://chenlab-data-public.s3.amazonaws.com/octad/octad.db/lincs_signatures.rda',
  'https://chenlab-data-public.s3.amazonaws.com/octad/octad.db/merged_gene_info.rda',
  'https://chenlab-data-public.s3.amazonaws.com/octad/octad.db/octad.LINCS.counts.rda',
  'https://chenlab-data-public.s3.amazonaws.com/octad/octad.db/phenoDF.rda',
  'https://chenlab-data-public.s3.amazonaws.com/octad/octad.db/random_gsea_score.rda',
  'https://chenlab-data-public.s3.amazonaws.com/octad/octad.db/tsne.rda',
  'https://chenlab-data-public.s3-us-west-2.amazonaws.com/octad/octad.counts.and.tpm.h5',
  'https://chenlab-data-public.s3.amazonaws.com/octad/octad.db/res_example.rda',
  'https://chenlab-data-public.s3.amazonaws.com/octad/octad.db/sRGES_example.rda'
),
SourceVersion=Sys.time(),
Species="Homo sapiens",
TaxonomyId='9606',
Coordinate_1_based=TRUE,
DataProvider='Michigan State University',
Maintainer='Eugene Chekalin <chekali1@msu.edu>',
RDataClass=c(
  'matrix array',
  'matrix array',
  'data.frame',
  'data.table data.frame',
  'matrix array',
  'list',
  'list',
  'list',
  'data.frame',
  'data.table data.frame',
  'matrix array',
  'data.table data.frame',
  'matrix array',
  'data.frame',
  'list',
  'data.frame',
  'data.frame',
  'data.frame',
  'data.frame'
),
DispatchClass=c(rep('Rda',16),'H5File',rep('Rda',2)),
RDataPath=c('octad.db/CCLE.log2.read.count.matrix.rda',
  'octad.db/CCLE.overlaps.rda',
  'octad.db/CCLE.sample.meta.rda',
  'octad.db/CTRPv2.sensprof.rda',
  'octad.db/EncoderDF.rda',
  'octad.db/cmpd_sets_ChemCluster.rda',
  'octad.db/cmpd_sets_chembl_targets.rda',
  'octad.db/cmpd_sets_mesh.rda',
  'octad.db/fda_drugs.rda',
  'octad.db/lincs_sig_info.rda',
  'octad.db/lincs_signatures.rda',
  'octad.db/merged_gene_info.rda',
  'octad.db/octad.LINCS.counts.rda',
  'octad.db/phenoDF.rda',
  'octad.db/random_gsea_score.rda',
  'octad.db/tsne.rda',
  'octad.counts.and.tpm.h5',
  'octad.db/res_example.rda',
  'octad.db/sRGES_example.rda'
),
Location_Prefix='https://chenlab-data-public.s3.amazonaws.com/octad/'
)
write.csv(meta, file="inst/extdata/metadata.csv", row.names=FALSE)